File: leetcode/746/solution.py
from tkinter import W
from typing import List, Tuple
from unittest import TestCase, main


class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        l = len(cost)
        for i in range(l - 1, -1, -1):
            if i < l - 2:
                cost[i] = min(cost[i + 1], cost[i + 2]) + cost[i]
        return min(cost[0], cost[1])


class Test(TestCase):
    s = Solution()
    data: List[Tuple[List[int], int]] = [
        ([10, 15, 20], 15),
        ([1, 100, 1, 1, 1, 100, 1, 1, 100, 1], 6),
    ]

    def test_solution(self):
        for input, expected in self.data:
            self.assertEqual(self.s.minCostClimbingStairs(input), expected)


if __name__ == "__main__":
    main()

File: leetcode/746/solution.rs
/**
 * https://leetcode.com/problems/min-cost-climbing-stairs/
 * 746. Min Cost Climbing Stairs
 */

impl Solution {
    pub fn min_cost_climbing_stairs(mut cost: Vec<i32>) -> i32 {
        for i in (0..(cost.len() - 2)).rev() {
            cost[i] += cost[i + 1].min(cost[i + 2]);
        }
        cost[0].min(cost[1])
    }
}
